javaparser-core/src/main/java/com/github/javaparser/ParseProblemException.java

```diff
@@ -0,0 +1,24 @@
+package com.github.javaparser;
+
+import java.util.List;
+
+/**
+ * Thrown when parsing problems occur during parsing with the static methods on JavaParser.
+ */
+public class ParseProblemException extends Exception {
```

Comment:
Nah, I think I agree, it's the Java 8 world and we want to use the functional programming features, and checked exceptions are hated there.